pkg/reconciler/route/config/domain.go

```diff
@@ -63,33 +69,50 @@ func (s *LabelSelector) Matches(labels map[string]string) bool {
 // Domain maps domains to routes by matching the domain's
 // label selectors to the route's labels.
 type Domain struct {
-	// Domains map from domain to label selector.  If a route has
-	// labels matching a particular selector, it will use the
-	// corresponding domain.  If multiple selectors match, we choose
-	// the most specific selector.
-	Domains map[string]*LabelSelector
+	// Domains map from domain to a set of options including a label selector.
+	Domains map[string]DomainConfig
+}
+
+// The configuration of one domain
+type DomainConfig struct {
+	// The label selector for the domain. If a route has labels matching a particular selector, it
+	// will use the corresponding domain. If multiple selectors match, we choose the most specific
+	// selector.
+	Selector *LabelSelector `json:"selector,omitempty"`
+	// If true, the domain will have a wildcard TLS certificate generated.
+	Wildcard bool `json:"wildcard"`
+}
+
+// Internal only representation of domain config for unmarshalling, allows backwards compatibility
+type domainInternalConfig struct {
+	Selector map[string]string `json:"selector,omitempty"`
+	Wildcard bool              `json:"wildcard"`
 }
 
 // NewDomainFromConfigMap creates a Domain from the supplied ConfigMap
 func NewDomainFromConfigMap(configMap *corev1.ConfigMap) (*Domain, error) {
-	c := Domain{Domains: map[string]*LabelSelector{}}
+	c := Domain{Domains: map[string]DomainConfig{}}
 	hasDefault := false
 	for k, v := range configMap.Data {
 		if k == configmap.ExampleKey {
 			continue
 		}
-		labelSelector := &LabelSelector{}
-		err := yaml.Unmarshal([]byte(v), labelSelector)
+		internalConfig := domainInternalConfig{}
+		err := yaml.Unmarshal([]byte(v), &internalConfig)
 		if err != nil {
 			return nil, err
 		}
-		c.Domains[k] = labelSelector
-		if len(labelSelector.Selector) == 0 {
+		if len(internalConfig.Selector) == 0 {
 			hasDefault = true
+			internalConfig.Wildcard = true
+		}
+		c.Domains[k] = DomainConfig{
+			Selector: &LabelSelector{Selector: internalConfig.Selector},
```

Comment:
nit: We could simplify LabelSelector now to be a type alias to `map[string]string`